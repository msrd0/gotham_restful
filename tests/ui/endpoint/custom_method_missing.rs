#[macro_use]
extern crate gotham_restful;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

#[endpoint(uri = "custom_read")]
async fn read_all() {}

fn main() {}
