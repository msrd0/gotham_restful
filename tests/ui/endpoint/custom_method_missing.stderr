error: Missing `method` attribute (e.g. `#[endpoint(method = "gotham_restful::gotham::hyper::Method::GET")]`)
 --> tests/ui/endpoint/custom_method_missing.rs:8:1
  |
8 | #[endpoint(uri = "custom_read")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `endpoint` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `read_all___gotham_restful_endpoint` in this scope
 --> tests/ui/endpoint/custom_method_missing.rs:5:12
  |
5 | #[resource(read_all)]
  |            ^^^^^^^^ not found in this scope
  |
help: you might be missing a type parameter
  |
4 | #[derive(Resource<read_all___gotham_restful_endpoint>)]
  |                  ++++++++++++++++++++++++++++++++++++
